src/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/HostWorkspace/LoadedProject.cs

```diff
@@ -54,13 +56,16 @@ public LoadedProject(ProjectSystemProject projectSystemProject, SolutionServices
         _projectDirectory = Path.GetDirectoryName(_projectFilePath)!;
 
         _fileChangeContext = fileWatcher.CreateContext([new(_projectDirectory, [".cs", ".cshtml", ".razor"])]);
-        _fileChangeContext.FileChanged += FileChangedContext_FileChanged;
+        _fileChangeContext.FileChanged += FileChangeContext_FileChanged;
+
+        _projectAssetsFileChangeContext = fileWatcher.CreateContext([]);
+        _projectAssetsFileChangeContext.FileChanged += ProjectAssetsFileChangeContext_FileChanged;
 
         // Start watching for file changes for the project file as well
         _fileChangeContext.EnqueueWatchingFile(_projectFilePath);
```

Comment:
The idea here of having two contexts is this can instead be done against your new context, and we can delete the first check in FileChangeContext_FileChanged too.